app/src/main/java/org/dicekeys/app/viewmodels/DiceKeyViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package org.dicekeys.app.viewmodels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import org.dicekeys.app.encryption.EncryptedDiceKey
+import org.dicekeys.app.encryption.EncryptedStorage
+import org.dicekeys.app.repositories.DiceKeyRepository
+import org.dicekeys.dicekey.DiceKey
+
+
+class DiceKeyViewModel @AssistedInject constructor(val encryptedStorage: EncryptedStorage, val diceKeyRepository: DiceKeyRepository, @Assisted val keyId: String) : ViewModel() {
+    val isSaved = MutableLiveData<Boolean>(false)
+    val diceKey = MutableLiveData<DiceKey<*>>()
```

Comment:
Trying to understand ViewModels on Android again.  Since the DiceKey is mutable and can have a nil value, this object exists at all times for the view and is just modified when a new DiceKey is loaded in, as opposed to reactive models where we'd just create a new backing state (model)?